core/domain/feedback_services.py

```diff
@@ -366,6 +366,34 @@ def _get_thread_from_model(thread_model):
         thread_model.created_on, thread_model.last_updated)
 
 
+def get_messages_for_exp_id(exp_id, limit=None):
+    """Fetch the feedback threads for the given exploration, and use those to
+    finally get the new (open) feedback messages.
+
+    Args:
+        exp_id: str.
+        limit: Max. number of FeedbackMessage objects to be returned. Default
+            value of None implies that the result will not be limited at all.
+
+    Returns:
+       A list of FeedbackMessage.
+    """
+    thread_models = feedback_models.FeedbackThreadModel.get_threads(exp_id)
+
+    message_models = []
+    for thread in thread_models:
+        message_models.extend(
+            feedback_models.FeedbackMessageModel.get_messages(
+                exp_id, thread.id, is_full_thread_id=True))
+
+    result = []
+    for message in message_models:
```

Comment:
Done. Also, we would've preferred to write "message" directly if it were a domain object, right ?